packages/catalog/src/core/utilities/parse-catalog-data/types.ts

```diff
@@ -9,18 +9,32 @@ export enum SortField {
   UpdatedAt = 'updatedAt',
   CreatedAt = 'createdAt',
   Title = 'title',
-  PublishedDate = 'publishDate',
+  PublishDate = 'publishDate',
   CourseStartDate = 'courseStartDate',
   Relevance = 'relevance'
 }
 
+export enum SortColumn {
+  CourseStartDate = 'courseStartDate',
+  CreatedAt = 'createdAt',
+  DisplayDate = 'displayDate',
+  Label = 'label',
+  LastActiveAt = 'lastActiveAt',
+  Name = 'name',
+  ParentName = 'parentName',
+  PublishDate = 'publishDate',
+  Relevance = 'relevance',
+  Title = 'title',
+  UpdatedAt = 'updatedAt'
+}
+
 export enum SortDirection {
   Desc = 'desc',
   Asc = 'asc'
 }
 
 export type Sort = {
-  field: SortField;
+  field: SortColumn;
   direction?: SortDirection;
 };
```

Comment:
Since the graphql layer is updated to use enums for both sort field and sort direction, alternatively we might just reference the generated typescript enums from `GlobalTypes` instead of creating our own here. There are some caveats with this approach though: seems that the TI catalog widget only uses a subset of the sort columns as declared in the graphql enum (referring to [this function in catalog widget](https://github.com/thoughtindustries/ti/blob/staging/assets/javascripts/lms/components/ti-browse-catalog.js#L9) which builds the sort dropdown). Was wondering if we want to do the same here in the catalog component. cc: @rjschill87

Reply:
@ljiang-ti Think referencing the enums from `GlobalTypes` sounds good. Not sure I'm following your concerns w/ the caveat though: the code you linked is for determining which sort options can be displayed to the user? May need you to expand on your thoughts there.